### ISSUE TITLE
Feedback Issue: JetBrains IDEs 💻 

### ISSUE THREAD

**Issue by loujaybee** [gitpod-io/gitpod#6576]
## Objective

This issue is to track feeback related to the JetBrains Desktop Remote Development Support. 

See related roadmap item [[1](https://github.com/gitpod-io/roadmap/issues/20)]

## Background

The JetBrains team has been hard at work [enabling remote development for the JetBrains editors](https://youtrack.jetbrains.com/issue/IDEA-226455#focus=Comments-27-5125731.0-0). This feature allows a user to use their Gitpod environment, whilst connecting using the JetBrains editors. 

Feel free to send in any comments, thoughts, emojis 🚀 👍  🎉 ❤️, screenshots, or screencasts as well as open any new separate issues with a description for further discussion. 💬 All known issues and feature requests will be labelled as [`editor: jetbrains`](https://github.com/gitpod-io/gitpod/labels/editor%3A%20jetbrains)

**Comment by loujaybee** [comment]
I think there was some confusion around the private beta for JetBrains ending and closing this issue, but since JetBrains is still in beta, I'll re-open and pin this issue 🙏

**Comment by akosyakov** [comment]
@loujaybee We need to update docs to point to this issue instead of closed one?

**Comment by tmy2017** [comment]
Hi gitpod! first thanks for your great work, gitpod really empowers me to participate/experiment with many open source projects. 

I have a question/feedback regarding JetBrains IDE desktop support (not sure is this issue the right place since it is so quiet here, but I also can not post on community.gitpod.io since there is in read-only now). 

How to restart the **Jetbrains IDE backend** manually and get the new CodeWithMe invitation link to reconnect to IDE (NOT restart the workspace itself -  #6679)? 

My use case is to install an IDE plugin and it requires me to restart IDE to take effect. But after I shutdown IDE, I am stuck, since I can not get the new code with me link. And if I manually "stop workspace" and restart, the IDE plugin is gone (I understand this is due to anything outside of /workspace is not persisted).

I tried to fiddle around with `/bin/bash -li /ide-desktop/startup.sh 24000 Open in PhpStorm` but to no avail. 

To conclude - before we have a full-grown solution to easily restart IDE backend or to persist IDE plugin, can anyone share a workaround on how to manually restart JetBrains IDE backend? ex: maybe kill/restart some processes or send some commands to `/.supervisor/supervisor`? 

(I have also tried Projector solution, but the plugin I am interested needs JCEF and currently projector does not fully support it yet - see [PRJ-285](https://youtrack.jetbrains.com/issue/PRJ-285), thus I need to use Desktop Jetbrains IDE)

**Comment by axonasif** [comment]
> but I also can not post on community.gitpod.io since there is in read-only now

Hi @tmy2017, just FYI: there is a channel specific to the jetbrains IDE at the [discord server](https://discord.gg/PwBUEzC96A)

**Comment by ChristianHuff-DEV** [comment]
Is there a way to open the IDE (Goland in my case) in a specific folder?

I have a mono repo that looks like this:
![Screenshot 2022-02-26 at 17 37 53](https://user-images.githubusercontent.com/46844303/155853346-81661d25-c116-482b-ad4f-aaf920ba6f52.png)

Only the "account-service" and "helpy" are Go projects. Other once (i.e. "webapp" is React) are not Go projects.

How can I open the IDE so that "account-service" is the root?

**Comment by loujaybee** [comment]
@ChristianHuff-DEV, yep I think this is the issue you'll want to track: https://github.com/gitpod-io/gitpod/issues/6900. Last time we checked, this wasn't possible yet with remote development, but when the API's become available, we'll be looking to support this use case, also 🙏

**Comment by gengjiawen** [comment]
Any progress on bring Clion ?

**Comment by emattiza** [comment]
As a user, when a workspace is running, I would like to open it from the jetbrains gateway without setting a user preference, or have a user preference that is multiple of the provided launch options. This would enable me to start a workspace without having to wait for preferences to cascade and then restart any new workspace with the appropriate remote link features enabled for the given ide of choice.

Thats the story anyways. Still love what you guys are doing and happy to work around it! But this could be polished in some way maybe?

**Comment by ENate** [comment]
Great job. But this gitpod plugin do not seem to work on Jetbrains Gateway version 2022.1.1 (build 221.5080.246). Besides, after installing the older version of Jetbrains Gateway (version 2021.3.2), it is still impossible to connect to the Gitpod workspace via SSH. Even after selecting the Intellij option from the workspace as described in the documentation.
Jetbrains Gateway 2022.1.1 do not open on Ubuntu 18.04.

**Comment by nooobcoder** [comment]
> Is there a way to open the IDE (Goland in my case) in a specific folder?
> 
> I have a mono repo that looks like this: ![Screenshot 2022-02-26 at 17 37 53](https://user-images.githubusercontent.com/46844303/155853346-81661d25-c116-482b-ad4f-aaf920ba6f52.png)
> 
> Only the "account-service" and "helpy" are Go projects. Other once (i.e. "webapp" is React) are not Go projects.
> 
> How can I open GoLand so that "account-service" is the root?
> 
> EDIT: Maybe an option could be added to the JetBrain Gateway plugin to not only open a workspace but also define the folder to open.

Exactly, I too have the same issue, @ChristianHuff-DEV I too need a solution for this to work with monorepos, for `code` this is not an issue, but with Jbr Gateway, it simple does not allow me to open a folder as the project root.

**Comment by frimtec** [comment]
I'm quite new to gitpod. As a addicted IntelliJ user I was pleased to hear that also Intellij is now supported - at least as beta.
I must say things looks really great.

Currently I have one problem, not sure if I do something wrong or if it is just a bug:
When I create a new workspace for a maven Java project, having a gitpod init task that does a "mvn install" or similar I would expect that the IntelliJ project is fully setup with all dependencies.
But that seams not to be true. The compiler cannot resolve any dependencies. So if I start a test, an application or just trigger a rebuild within the IDE, I get a huge list of compiler errors (see screenshot).
![gitpod-maven](https://user-images.githubusercontent.com/3511114/175807680-03c73899-9957-4913-a750-e440876b1a7a.png)

The only workaround to resolve from this state is:
- Do a mvn install from within IntelliJ UI (maven tab).
- Reload the maven modules within IntelliJ UI (maven tab).
- Rebuild the project within IntelliJ UI. 

This problem can be reproduced with my repo https://github.com/frimtec/jpse and also with https://github.com/frimtec/my-java-template an instance from the gitpod java template.

**Comment by jeanphibaconnais** [comment]
Hi, i'm started to test Jetbrains Gateway and with the latest version when I open a workspace I have this error 
``` 
'com.intellij.ssh.ConnectionBuilder com.intellij.ssh.RemoteCredentialsUtil.connectionBuilder$default(com.intellij.remote.RemoteCredentials, com.intellij.openapi.project.Project, com.intellij.openapi.progress.ProgressIndicator, boolean, int, java.lang.Object)'
``` 

Have you ever seen that ?


JetBrains Gateway 2022.2
Build #GW-222.3345.108, built on July 22, 2022
Runtime version: 17.0.3+7-b469.32 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 12.4

**Comment by huweiATgithub** [comment]
I had an issue regarding the loading of user's `.bashrc`  configuration: https://github.com/gitpod-io/gitpod/issues/13815

**Comment by andreafalzetti** [comment]
Test notification

**Comment by andreafalzetti** [comment]
test

**Comment by andreafalzetti** [comment]
test 3

**Comment by BA447** [comment]
Php

**Comment by BA447** [comment]
Ggg

**Comment by krzysztofKubiakVizlib** [comment]
Hi, did you try to solve an issue with the terminal in the JetBrains app when Gitpod was in use? The console is narrow and short (only part of the red border is in use) and has a lot of unused space. 

![Screenshot 2023-04-07 at 09 52 06](https://user-images.githubusercontent.com/66253914/230567106-a9b9cd31-b958-4a4d-9fa4-0065efde40bf.png)

**Comment by perkinss** [comment]
The "Open in RubyMine" button doesn't work for me at all.

When I inspect the button it looks like it's not connected with any actions. 
<img width="304" alt="image" src="https://github.com/gitpod-io/gitpod/assets/405204/fe5e6a29-1825-4f4b-81ca-7fb6edd806ae">

**Comment by akosyakov** [comment]
@perkinss There should be instruction just above the button 👀 but maybe we can make it somehow more prominent cc @gtsiolis

**Comment by sjoshid** [comment]
Any way to use a specific version of intellij when starting a workspace? I have a fallback license I'd like to use.

**Comment by axonasif** [comment]
Hi @sjoshid, you can pin a version from https://gitpod.io/settings/policy > Available editors

**Comment by sjoshid** [comment]
> Hi @sjoshid, you can pin a version from https://gitpod.io/settings/policy > Available editors

I dont have a intellij fallback license for any of these.  
![image](https://github.com/user-attachments/assets/8d35d724-a74d-4e3a-9d5c-70100914ef91)

I paid for 2024.2.3 and pinned it. I really hope I dont have to pay again next year because Gitpod won't support it.

**Comment by axonasif** [comment]
@sjoshid apologies. Which version did you have a license for?

**Comment by sjoshid** [comment]
> @sjoshid apologies. Which version did you have a license for?

23.1

**Comment by lwoerdeman** [comment]
Hi Gitpod team, I was wondering what the release cadence of the IntelliJ gitpod client is and if this is published anywhere in the documentation - it seems to lag slightly behind the IntelliJ stable release cycle (for example currently using 2024.3 - even though latest version of the stable client is 2024.3.1.1)

**Comment by mustard-mh** [comment]
Good day @lwoerdeman 👋 , the stable IntelliJ 2024.3.1.1 has deployed on Gitpod Classic.

> I was wondering what the release cadence of the IntelliJ gitpod host is and if this is published anywhere in the documentation

Gitpod has a [GitHub Action](https://github.com/gitpod-io/gitpod/blob/main/.github/workflows/jetbrains-updates.yml) to trigger stable releases, developer will review and merge if the integration tests + smoke tests are passed and no breaking changes or compatible issues. 

|Img2|Img2|
|:-:|:-:|
|<img width="892" alt="image" src="https://github.com/user-attachments/assets/6b94360c-31ee-4d28-8596-b80a97eac009" />|<img width="965" alt="image" src="https://github.com/user-attachments/assets/355b9d5e-72b5-4eca-a097-53d86511940f" />|
